### PR TITLE
[SessionD] Do not report credit that is in final state

### PR DESCRIPTION
**Pull request by themarwhal**
Signed-off-by: Marie Bremner <marwhal@fb.com>

## Summary
A bit of an edge case previously not covered by any tests.
When we have a credit that is final, we do NOT want to report any usage because there is no credit to be given. We will only send an update request if it is prompted by a Gy ReAuth. 

Also added a unit test to cover this case. The test fails without the change.

## Test Plan
SessionD unit tests
CWF integ test + manually verifying on the setup with large delays between traffic chunks to make sure there are no updates

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

File: lte/gateway/c/session_manager/ChargingGrant.cpp [hunk 1]
```diff
@@ -136,7 +136,7 @@ bool ChargingGrant::get_update_type(
     *update_type = CreditUsage::REAUTH_REQUIRED;
     return true;
   }
-  if (is_final_grant && credit.is_quota_exhausted(1)) {
+  if (is_final_grant) {
```
Comment (uri200) [inline lte/gateway/c/session_manager/ChargingGrant.cpp:139]: 😍

**Review by uri200 — APPROVED** [review]
